research/src/pages/file/file.proposal.mdx

```diff
@@ -0,0 +1,212 @@
+| [`multiple`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/multiple)   | `boolean`                                                                 | `false`          | A `boolean` which, if present, indicates that the user may choose more than one file                                                                                        |
+
+## Methods
+
+N/A
+
+## Events
+
+| Event Name | Detail Type | Bubbles | Composed | Cancellable | Dispatch Behavior                                       |
+| ---------- | ----------- | ------- | -------- | ----------- | ------------------------------------------------------- |
+| `change`   | none        | `true`  | `true`   | `false`     | Fired when the input's `value` is commited by the user. |
+| `input`    | none        | `true`  | `true`   | `false`     | Fired when the input's `value` changes.                 |
+
+### Anatomy
+
+#### Diagram
+
+<FileAnatomy />
+
+#### DOM Structure
+
+```html
+<host>
+  <div part="file-selector-button">
+    <slot name="file-selector-button"></slot>
+  </div>
+  <div part="label">
+    <slot name="label"></slot>
+  </div>
+</host>
+```
+
+#### Slots
+
+| Slot Name              | Description                                                                           | Fallback Content              |
+| ---------------------- | ------------------------------------------------------------------------------------- | ----------------------------- |
+| `file-selector-button` | Invokes the file selection prompt.                                                    | A button element is provided. |
+| `label`                | Indicates if a file input is waiting for file selection, or if it has selected files. | A label element is provided.  |
+
+#### CSS Parts
+
+| Part Name              | Description                                           |
+| ---------------------- | ----------------------------------------------------- |
+| `file-selector-button` | The controls button.                                  |
+| `label`                | The controls label, indicating file selection status. |
+
+---
+
+## Behavior
+
+### States and Interactions
+
+When the input is invoked, the file selection prompt will limit the user to
+selecting files that match the `accept` attribute, and the user will be limited
+to selecting a single file unless the `multiple` attribute is `true`.
+
+When a file are selected, the `label` updates to show the filename. When set to
+`multiple`, the number of selected files is shown instead of a filename.
+
+If a user cancels the prompt (choosing not to select a file), the input shows
+the default (no selection) state. This also clears any previously selected
+files.
+
+#### Drag and drop
+
+Users may also drag and drop files onto the input, where the drop target is the
+containing element.
+
+#### Accessibility
+
+The file input should be focusable via keyboard navigation.
```

Comment:
can you file an issue as this should be more clear on what "keyboard navigation" means. Likewise we'll want to understand if there are any name computation adjustments that should be made (eg: on change, etc)

Reply:
Perhaps we drop this statement? Really this just means that the input has is tabbable, and I wouldn't expect it to blur on change (aligning with other controls).